From 9b033baa51a129605251b05a8225807e39ab12e2 Mon Sep 17 00:00:00 2001
Message-Id: <9b033baa51a129605251b05a8225807e39ab12e2.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Gaurav K Singh <gaurav.k.singh@intel.com>
Date: Tue, 13 Jan 2015 21:43:31 +0530
Subject: [PATCH 39/95] MUST_REBASE [VPG]: video/adf/intel: Generalize DSI
 regs initialization

Generalizing the DSI regs offsets initialization for single link panels
and dual link panels.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-5276
Change-Id: I2791949bc3fbdefae0a12ac443479235824b9f68
Signed-off-by: Gaurav K Singh <gaurav.k.singh@intel.com>
---
 drivers/video/adf/intel/core/vlv/vlv_dc_config.c |   11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/drivers/video/adf/intel/core/vlv/vlv_dc_config.c b/drivers/video/adf/intel/core/vlv/vlv_dc_config.c
index d97fe5f..6812a9e 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_dc_config.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_dc_config.c
@@ -230,12 +230,19 @@ static int vlv_initialize_port(struct vlv_dc_config *vlv_config,
 	struct vlv_dsi_port *dsi_port = NULL;
 	struct vlv_hdmi_port *hdmi_port = NULL;
 	struct vlv_pipeline *disp = NULL;
+	struct dsi_pipe *dsi_pipe = NULL;
+	struct dsi_context *dsi_ctx = NULL;
+	enum port port_no;
 
 	disp = &vlv_config->pipeline[disp_no];
 	switch (type) {
 	case INTEL_PIPE_DSI:
-		dsi_port = &disp->port.dsi_port[port];
-		vlv_dsi_port_init(dsi_port, port, pipe);
+		dsi_pipe = &disp->gen.dsi;
+		dsi_ctx = &dsi_pipe->config.ctx;
+		for_each_dsi_port(port_no, dsi_ctx->ports) {
+			dsi_port = &disp->port.dsi_port[port_no];
+			vlv_dsi_port_init(dsi_port, port, pipe);
+		}
 		break;
 	case INTEL_PIPE_HDMI:
 		hdmi_port = &disp->port.hdmi_port;
-- 
1.7.9.5

